### PR TITLE
nixos/openvpn: restart client after suspend

### LINKED ISSUE #52528: OpenVPN service does not reconnect from suspend
## Issue description
services.openvpn.* creates a usable VPN connection but only when manually {re}starting the service. On my Purism 13 laptop, both a cold boot fails with `start request repeated too quickly` and when the service attempts to reconnect after a suspend. The service never attempts to reconnect from a failed state unless I manually intervene.

## Technical details

When woken from suspend, `systemctl status openvpn-{name}.service` reports:
```
Dec 18 21:59:26 arha systemd[1]: openvpn-windscribe.service: Service RestartSec=100ms expired, scheduling restart.
Dec 18 21:59:26 arha systemd[1]: openvpn-windscribe.service: Scheduled restart job, restart counter is at 5.
Dec 18 21:59:26 arha systemd[1]: Stopped OpenVPN instance ‘windscribe’.
Dec 18 21:59:26 arha systemd[1]: openvpn-windscribe.service: Start request repeated too quickly.
Dec 18 21:59:26 arha systemd[1]: openvpn-windscribe.service: Failed with result 'exit-code'.
Dec 18 21:59:26 arha systemd[1]: Failed to start OpenVPN instance ‘windscribe’.
```
My configuration:
```
    openvpn.servers = {
          windscribe = {
            config = ''
              client
              dev tun
              proto tcp
              remote pl.windscribe.com 443

              nobind

              --auth-user-pass /home/taohansen/windscribe-pass.txt
              auth-nocache

              resolv-retry infinite

              auth SHA512
              cipher AES-256-CBC
              comp-lzo
              verb 2
              mute-replay-warnings
              remote-cert-tls server
              persist-key
              persist-tun

              key-direction 1

              <ca>
              -----BEGIN CERTIFICATE-----
              {ca}
              -----END CERTIFICATE-----
              </ca>
              <tls-auth>
              -----BEGIN OpenVPN Static key V1-----
              {tls}
              -----END OpenVPN Static key V1-----
              </tls-auth>
            '';
            autoStart = true;
            updateResolvConf = true;
            };
          };
```

### PR DESCRIPTION
**Pull request by gbtb**
###### Description of changes

Openvpn clients do not handle sleep and resume of NixOS correctly and hang in "zombie" state until manually restarted.
This PR adds a new option to the openvpn module and a simple systemd unit that sends a signal to openvpn process, which in turn makes openvpn process self-close and therefore trigger a restart of the openvpn unit by systemd.

[Man page about sleep.target](https://man.archlinux.org/man/systemd-sleep.8.en)
[Wiki page with example of restart unit](https://wiki.archlinux.org/title/OpenVPN#Client_daemon_not_reconnecting_after_suspend)

Resolves https://github.com/NixOS/nixpkgs/issues/52528

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/openvpn.nix [hunk 1]
```diff
@@ -70,6 +71,17 @@ let
       serviceConfig.Type = "notify";
     };
 
+  restartService =
```
Comment (SuperSandro2000) [inline nixos/modules/services/networking/openvpn.nix:74]: Please use optionalAttrs

File: nixos/modules/services/networking/openvpn.nix [hunk 2]
```diff
@@ -70,6 +71,17 @@ let
...
+    if (cfg.restartAfterSleep) then {
+      openvpn-restart =
+        {
```
Comment (SuperSandro2000) [inline nixos/modules/services/networking/openvpn.nix:77]:
```suggestion
      openvpn-restart = {
```

**Comment by gbtb** [comment]
Hello, @SuperSandro2000. Could you look into my PR once again please? I've made changes you've suggested.

**Comment by gbtb** [comment]
Sorry for being annoying :sweat_smile: , @SuperSandro2000, but I'd really like to have this merged into master ...

**Comment by SuperSandro2000** [comment]
no problem, I have to many github notifcations and sometimes miss things.

**Pull request merged**